src/Npgsql/Replication/NpgsqlReplicationConnection.cs

```diff
@@ -420,10 +420,10 @@ internal async Task<NpgsqlReplicationSlotOptions> CreateReplicationSlotInternal(
                 break;
             case BackendMessageCode.CommandComplete:
             {
-                EnsureAndSetState(ReplicationConnectionState.Idle);
                 yield break;
             }
             default:
+                stateResetter.ChangeResetState(ReplicationConnectionState.Broken);
```

Comment:
Is this actually necessary? The resetter checks if the connector is broken, which the next line will do, no?

Reply:
No, it isn't

Reply:
https://github.com/Brar/npgsql/pull/11